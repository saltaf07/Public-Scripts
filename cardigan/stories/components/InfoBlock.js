import { storiesOf } from '@storybook/react';
import InfoBlock from '../../../common/views/components/InfoBlock/InfoBlock';
import Readme from '../../../common/views/components/InfoBlock/README.md';

const InfoBlockExample = () => {
  return (
    <InfoBlock
      title={'Book your ticket'}
      text={[
        {
          type: 'list-item',
          text:
            'It’s still free to visit our museum and library. You’ll just need to choose a time slot and book a ticket before you arrive.',
          spans: [],
        },
        {
          type: 'list-item',
          text:
            'You can explore the museum for as long as you’d like, but  library sessions are now limited to either a morning or an afternoon. Library tickets include museum entry too.',
          spans: [],
        },
      ]}
      linkText={`Book your ticket`}
      link={{ link_type: 'Web', url: `/covid-book-your-ticket` }}
    />
  );
};

const stories = storiesOf('Components', module);
stories.add('InfoBlock', InfoBlockExample, { readme: { sidebar: Readme } });
