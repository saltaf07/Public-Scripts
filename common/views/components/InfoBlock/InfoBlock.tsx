import Space from '@weco/common/views/components/styled/Space';
import { classNames } from '@weco/common/utils/classnames';
import PrismicHtmlBlock from '@weco/common/views/components/PrismicHtmlBlock/PrismicHtmlBlock';
import { HTMLString } from '../../../services/prismic/types';
import ButtonOutlinedLink from '@weco/common/views/components/ButtonOutlinedLink/ButtonOutlinedLink';

type Props = {
  heading: string;
  text: HTMLString;
  linkText: string | null;
  link: any | null; // TODO: use PrismicLink type from #5636 and parseUrl
};

const InfoBlock = ({ heading, text, linkText, link }) => {
  return (
    <Space
      h={{ size: 'l', properties: ['padding-left', 'padding-right'] }}
      className={classNames({
        'border-color-yellow': true,
      })}
      style={{ borderLeftWidth: '16px', borderLeftStyle: 'solid' }}
    >
      <h2 className="h2">{heading}</h2>
      <div className="spaced-text body-text">
        <PrismicHtmlBlock html={text} />
      </div>
      {link && (
        <Space v={{ size: 'l', properties: ['margin-top'] }}>
          <ButtonOutlinedLink link={link.url} text={linkText} />
        </Space>
      )}
    </Space>
  );
};

export default InfoBlock;
