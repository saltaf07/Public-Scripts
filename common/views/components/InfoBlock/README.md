## Purpose

To call out important information on a page, potentially linking off to other relevant content.

[Edit this on GitHub](https://github.com/wellcomecollection/wellcomecollection.org/edit/master/common/views/components/InfoBlock/README.md)
