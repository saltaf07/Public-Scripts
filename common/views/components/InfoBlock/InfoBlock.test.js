import InfoBlock from './InfoBlock';
import { shallowWithTheme } from '../../../test/fixtures/enzyme-helpers';

const title = `InfoBlock title`;
const text = [
  {
    type: 'paragraph',
    text: 'Text in the InfoBlock',
    spans: [],
  },
];
const link = { link_type: 'Web', url: 'https://wellcomecollection.org' };
const linkText = 'InfoBlock link text';
describe('InfoBlock', () => {
  it(`should match snapshots`, () => {
    const component = shallowWithTheme(
      <InfoBlock title={title} text={text} linkText={linkText} link={link} />
    );
    const componentNoLinkText = shallowWithTheme(
      <InfoBlock title={title} text={text} linkText={null} link={link} />
    );
    const componentNoLink = shallowWithTheme(
      <InfoBlock title={title} text={text} linkText={linkText} link={null} />
    );
    const componentNoLinkNoLinkText = shallowWithTheme(
      <InfoBlock title={title} text={text} linkText={null} link={null} />
    );

    expect(component.html()).toMatchSnapshot();
    expect(componentNoLink.html()).toMatchSnapshot();
    expect(componentNoLinkText.html()).toMatchSnapshot();
    expect(componentNoLinkNoLinkText.html()).toMatchSnapshot();
  });
});
