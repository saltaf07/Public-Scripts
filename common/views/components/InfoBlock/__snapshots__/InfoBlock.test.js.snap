// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBlock should match snapshots 1`] = `"<div class=\\"Space-sc-4az8w3-0 ivnzzq border-color-yellow\\" style=\\"border-left-width:16px;border-left-style:solid\\"><h2 id=\\"infoblock-title\\" class=\\"h2\\">InfoBlock title</h2><div class=\\"spaced-text body-text\\"><p>Text in the InfoBlock</p></div><div class=\\"Space-sc-4az8w3-0 iyqXRj\\"><a href=\\"https://wellcomecollection.org\\" class=\\"ButtonSolid__BaseButton-is2j2c-0 ButtonOutlined__OutlinedButton-sc-21smqz-0 fSydKO link-reset\\"><span class=\\"ButtonSolid__BaseButtonInner-is2j2c-1 dxoWEE font-hnm font-size-5  flex flex--v-center\\">InfoBlock link text</span></a></div></div>"`;

exports[`InfoBlock should match snapshots 2`] = `"<div class=\\"Space-sc-4az8w3-0 ivnzzq border-color-yellow\\" style=\\"border-left-width:16px;border-left-style:solid\\"><h2 id=\\"infoblock-title\\" class=\\"h2\\">InfoBlock title</h2><div class=\\"spaced-text body-text\\"><p>Text in the InfoBlock</p></div></div>"`;

exports[`InfoBlock should match snapshots 3`] = `"<div class=\\"Space-sc-4az8w3-0 ivnzzq border-color-yellow\\" style=\\"border-left-width:16px;border-left-style:solid\\"><h2 id=\\"infoblock-title\\" class=\\"h2\\">InfoBlock title</h2><div class=\\"spaced-text body-text\\"><p>Text in the InfoBlock</p></div></div>"`;

exports[`InfoBlock should match snapshots 4`] = `"<div class=\\"Space-sc-4az8w3-0 ivnzzq border-color-yellow\\" style=\\"border-left-width:16px;border-left-style:solid\\"><h2 id=\\"infoblock-title\\" class=\\"h2\\">InfoBlock title</h2><div class=\\"spaced-text body-text\\"><p>Text in the InfoBlock</p></div></div>"`;
